### DISCUSSION TITLE
IDEA: assertk-json extension library

### DISCUSSION THREAD

**Discussion by seepick** [assertk-org/assertk#348] [Ideas]
hi, very often i am in need of testing some JSON response, whereas i don't care about whitespace and ordering of keys within the JSON (semantic comparison).
there is the [JSONAssert](https://github.com/skyscreamer/JSONassert) library, which can do that. and sometimes i need to "dig into" the JSON, which can be done with [json-path](https://github.com/json-path/JsonPath).

e.g. transforming `JSONAssert.assertEquals(expectedJson, actualJson, jsontMode)` into `fun Assert<String>.isJsonEquals(expectedJson: String, mode: JsonMode = JsonMode.Strict)`

my idea would be to create another artifact, depending on the "assertk-core" library, and simply adding a transitive dependency with such a thin layer making the libs reusable via the fluent assertk assertions.

WDTY?

**Comment by evant** [comment]
Seems like a good idea. We've been planning an android extension lib in a similar vein for a while, but haven't had the time for it. If you want to work on it, I'd be glad to help review the design.

**Comment by evant** [comment]
One thing to note: right now assertk is multiplatform and it would be cool for a json artifact to be as well. But the libs you linked to are java only. Re-implementing in kotlin would be a fair bit of work though if those don't already exist so understand if you wouldn't want to go down that route.

**Comment by seepick** [comment]
i wasn't aware of the fact that assertk is multiplatform, until my first PR ;) very sophisticated, nice job.

and yes, you are right, it seems a bit much to re-implement a proper json assertion functionality from scratch... have to check the licenses and maybe just copy'n'paste the relevant stuff might be enough?!

how about "to help review the design"? where can i be of support exactly?

**Comment by evant** [comment]
I meant I can help review prs/help with any discussions on what the api should look like, if you were interested in implementing this feature.

**Comment by ghost** [comment]
I also needed json assertions. So while json assertion is not implemented in assertk core, I would like to share my assertk-jvm-json extension for JVM platform https://github.com/themichailov/assertk-jvm-json.
When json comparison will be implemented in assertk core for multiplatform, I will deprecate my lib.
